arch/xtensa/Kconfig

```diff
@@ -243,6 +243,14 @@ config XTENSA_MPU_DEFAULT_MEM_TYPE
 	  If userspace is enabled, it will be used to restore the memory type of
 	  the region being removed from a memory domain.
 
+config XTENSA_MPU_ONLY_SOC_RANGES
+	bool
+	help
+	  Enable this by the SoC to indicate to the architecture code to use
+	  the MPU ranges specified by SoC only, and skip the common ranges
+	  defined in the core architecture code. This gives total control to
+	  the SoC on the MPU ranges.
```

Comment:
This seems like needless permissions management?  SOC authors aren't malicious actors who can't be trusted with core memory space management, they're... us.  What is the =n case trying to fix or prevent?  Nothing wrong with setting up stuff at the arch layer and letting the SOC override however it wants.

Reply:
This simply is to give the SoC layer developers more flexibility to define their own map instead of relying on the arch layer for common regions. With 4K alignment, sometimes there are lots of wasted space between sections. So if SoC developers want, they can consolidate some regions together to get rid of those empty space. With MMU, hardware should have been designed with lots of memory since the page tables take up quite a bit. But with MPU, since there is no need for reserved memory for page tables, the hardware may have little memory to being with. Hence the need for some flexibility here.